www-apps/hugo/hugo-0.109.0.ebuild

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit go-module bash-completion-r1
+
+DESCRIPTION="Fast static HTML and CSS website generator"
+HOMEPAGE="https://gohugo.io https://github.com/gohugoio/hugo"
+SRC_URI="
+	https://github.com/gohugoio/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://tastytea.de/files/gentoo/${P}-vendor.tar.xz
+"
+
+# NOTE: To create the vendor tarball, run:
+# `go mod vendor && cd .. && tar -caf ${P}-vendor.tar.xz ${P}/vendor`
+
+LICENSE="Apache-2.0 BSD BSD-2 MIT Unlicense"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~loong ~x86"
+IUSE="doc +sass test"
+
+BDEPEND="
+	>=dev-lang/go-1.18
+	test? (
+		dev-python/docutils
+		dev-ruby/asciidoctor
+		virtual/pandoc
+	)
+"
+RDEPEND="
+	>=media-libs/libwebp-1.2.3-r1:=
+	sass? ( dev-libs/libsass:= )
+"
+DEPEND="${RDEPEND}"
+
+RESTRICT="!test? ( test )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.96.0-unbundle-libwebp-and-libsass.patch
+	"${FILESDIR}"/${PN}-0.109.0-skip-some-tests.patch
+	"${FILESDIR}"/${PN}-0.99.1-test-timeout.patch
+)
+
+src_configure() {
+	export CGO_ENABLED=1
+	export CGO_CFLAGS="${CFLAGS}"
+	export CGO_CPPFLAGS="${CPPFLAGS}"
+	export CGO_CXXFLAGS="${CXXFLAGS}"
+	export CGO_LDFLAGS="${LDFLAGS}"
+
+	MY_BUILDFLAGS="$(usev sass "-tags extended")"
+
+	default
+}
+
+src_compile() {
+	mkdir -pv bin || die
+	ego build -ldflags "-X github.com/gohugoio/hugo/common/hugo.vendorInfo=gentoo:${PVR}" \
+		${MY_BUILDFLAGS} -o "${S}/bin/hugo"
+
+	bin/hugo gen man --dir man || die
+
+	mkdir -pv completions || die
+	bin/hugo completion bash > completions/hugo || die
+	bin/hugo completion fish > completions/hugo.fish || die
+	bin/hugo completion zsh > completions/_hugo || die
+
+	if use doc ; then
+		bin/hugo gen doc --dir doc || die
+	fi
+}
+
+src_test() {
+	ego test -race "./..." ${MY_BUILDFLAGS}
+}
```

Comment:
```
>>> Test phase: www-apps/hugo-0.109.0
go test -race ./... -tags extended
-buildmode=pie not supported when -race is enabled
 * ERROR: www-apps/hugo-0.109.0::gentoo failed (test phase)
```

No idea where the buildmode comes from, but you may have to strip flags when testing the package.

Reply:
Hmm, i always get `-buildmode=exe` and neither go-module.eclass nor the ebuild set `-buildmode=pie`. I'm using 1.19.4 but it doesn't look like it changed [in 1.19.5](https://github.com/golang/go/issues?q=milestone%3AGo1.19.5+label%3ACherryPickApproved)…

I guess i'll set `-buildmode=exe` explicitly for tests for now. :thinking:

Reply:
I can't reproduce the pie-problem, but forced `-buildmode=exe` for tests and hope it'll fix it. 🤞

Reply:
`-buildmode=pie` came from the eclass, by the way. → <https://bugs.gentoo.org/702598>  
I couldn't reproduce because my tree was too old.

Reply:
It's better to just drop -race IMO. It's almost like using sanitizers or valgrind. It's not really appropriate for tests in ebuilds.